### PR TITLE
Added more unit tests

### PR DESCRIPTION
**Pull request by Marketionist**
**This pull request adds more unit tests**

Also:
- ```calendar.html``` was updated for usage in unit testing

### CODE REVIEW

File: aldryn_events/tests/frontend/fixtures/calendar.html [hunk 1]
```diff
@@ -1,7 +1,57 @@
...
     <div class="controls">
-        <div class="js-trigger">
-            <div data-direction="next"></div>
-        </div>
+        <div class="js-trigger"></div>
     </div>
 </div>
+
+<div class="events-calendar js-events-calendar" data-url="/en/events/get-dates/" data-pk="" data-error="There was a problem accessing the calendar, please try again.">
+
```
Comment (FinalAngel) [inline aldryn_events/tests/frontend/fixtures/calendar.html]: remove the additional addons and keep to 120 lines

File: aldryn_events/tests/frontend/unit/test.cl.events.js [hunk 2]
```diff
@@ -10,7 +10,7 @@
     /* global Cl, describe, it, expect */
 
     // ########################################################################
-    describe('cl.events.js', function () {
+    describe('cl.events.js:', function () {
```
Comment (FinalAngel) [inline aldryn_events/tests/frontend/unit/test.cl.events.js]: do we need the colon? The sentence should read "cl.events.js has available Cl namespace"?

File: aldryn_events/tests/frontend/unit/test.cl.events.js [hunk 3]
```diff
@@ -47,6 +48,34 @@
                 { preventDefault: function () {} })
             ).toEqual(false);
         });
+
+        it('Cl.events._handler returns undefined if direction is "next" and month is 7', function () {
+            expect(Cl.events._handler.call(
+                $('.js-trigger')[1],
+                { preventDefault: function () {} })
```
Comment (FinalAngel) [inline aldryn_events/tests/frontend/unit/test.cl.events.js]: I'd suggest following the DRY principle, for example you could write:

```
var noop = function () {};

{ preventDefault: noop })
```

or go even further:

```
var prevent = { preventDefault: function () {} }
expect(Cl.events._handler.call(
    $('.js-trigger')[2], prevent).toEqual(undefined);
```

File: aldryn_events/tests/frontend/unit/test.cl.events.js [hunk 4]
```diff
@@ -47,6 +48,34 @@
                 { preventDefault: function () {} })
             ).toEqual(false);
         });
+
+        it('Cl.events._handler returns undefined if direction is "next" and month is 7', function () {
```
Comment (vxsx) [inline aldryn_events/tests/frontend/unit/test.cl.events.js]: function returns undefined if there's a direction, everything else is not really needed
what you should test instead is that the $.ajax was called with correct parameters given different input

File: aldryn_events/tests/frontend/unit/test.cl.events.js (hunk 2 above)
Comment (Marketionist) [inline aldryn_events/tests/frontend/unit/test.cl.events.js]: We need a colon to make the sentence more universal - for example: "cl.events.js: Cl.events._handler returns undefined if direction is "previous" and month is 1"

File: aldryn_events/tests/frontend/fixtures/calendar.html [hunk 5]
```diff
@@ -1,7 +1,57 @@
...
     <div class="controls">
-        <div class="js-trigger">
-            <div data-direction="next"></div>
-        </div>
+        <div class="js-trigger"></div>
     </div>
 </div>
+
+<div class="events-calendar js-events-calendar"
+data-url="/en/events/get-dates/" data-pk="" data-error="There was a problem
```
Comment (FinalAngel) [inline aldryn_events/tests/frontend/fixtures/calendar.html]: there you'd need to add 4 space indentation (on all following instances)

File: aldryn_events/tests/frontend/unit/test.cl.events.js (hunk 2 above)
Comment (FinalAngel) [inline aldryn_events/tests/frontend/unit/test.cl.events.js]: just saying feel free to change it, you can also consult with @evildmp to create the structure. In the end it needs to be readable

**Comment by vxsx** [comment]
looks good

**Comment by FinalAngel** [comment]
LGTM but please still get the advise from @evildmp regarding the sentence structure

**Comment by Marketionist** [comment]
Ok, will ask and then add to "best practices"

**Pull request merged**